config/storage/resource-store/dynamic.json

```diff
@@ -14,14 +14,17 @@
       "@id": "urn:solid-server:default:ResourceStore",
       "@type": "MonitoringStore",
       "MonitoringStore:_source": {
-        "@id": "urn:solid-server:default:ResourceStore_Locking",
-        "LockingResourceStore:_source": {
-          "@id": "urn:solid-server:default:ResourceStore_Patching",
-          "PatchingStore:_source": {
-            "@id": "urn:solid-server:default:ResourceStore_Converting",
-            "RepresentationConvertingStore:_source": {
-              "@id": "urn:solid-server:default:ResourceStore_Routing",
-              "comment": "A router rule will be defined when activating dynamic pods."
+        "@id": "urn:solid-server:default:ResourceStore_Index",
+        "IndexRepresentationStore:_source": {
+          "@id": "urn:solid-server:default:ResourceStore_Locking",
+          "LockingResourceStore:_source": {
+            "@id": "urn:solid-server:default:ResourceStore_Patching",
+            "PatchingStore:_source": {
+              "@id": "urn:solid-server:default:ResourceStore_Converting",
+              "RepresentationConvertingStore:_source": {
+                "@id": "urn:solid-server:default:ResourceStore_Routing",
+                "comment": "A router rule will be defined when activating dynamic pods."
+              }
```

Comment:
The only way I see an array working is if the source is being set by a separate function after instantiation.

Linked list would indeed solve the indentation issue, would make the configs a bit larger though.

My main idea was to have a `config/storage/middleware` and a `config/storage/backend` folder (instead of the `config/storage/resource-store` folder we have now). The first would then contain the monitoring -> index -> locking -> patching -> converting chain, which is the same for all the different store configs we currently have. The second one would then have the routing and data accessor parts. 

If a new store needs to get added (such as for shapetrees for example), only 1 file would have to be editted.